src/normal.cc

```diff
@@ -189,9 +189,24 @@ String build_autoinfo_for_mapping(const Context& context, KeymapMode mode,
             descs.emplace_back(std::move(keys), built_in.docstring);
     }
 
+    // To combine mapped keys with the same command and docstrings (synomyms),
+    // we need a map (command, docstring) -> mapped keys
+    HashMap<std::pair<String, String>, Vector<String>> synonyms;
     for (auto& key : keymaps.get_mapped_keys(mode))
-        descs.emplace_back(key_to_str(key),
-                           keymaps.get_mapping(key, mode).docstring);
+    {
+        KeymapManager::KeymapInfo key_info = keymaps.get_mapping(key, mode);
+        std::pair<String, String> cmd_and_docstr = {join(key_info.keys | transform(key_to_str)),
```

Comment:
I think you can do `accumulate(..., String{}, std::plus{})` in place of join here, avoiding the need to define a new one in string_utils.